server/src/game/components/cult.rs

```diff
@@ -0,0 +1,110 @@
+use crate::game::{chat::{ChatGroup, ChatMessageVariant}, phase::PhaseType, player::PlayerReference, role::{apostle::Apostle, disciple::Disciple, zealot::Zealot, RoleState}, role_list::Faction, Game};
+
+
+impl Game {
+    pub fn cult(&self)->&Cult{
+        &self.cult
+    }
+    pub fn set_cult(&mut self, cult: Cult){
+        self.cult = cult;
+    }
+}
+#[derive(Default, Clone)]
+pub struct Cult {
+    pub ordered_cultists: Vec<PlayerReference>,
+    pub sacrifices_needed: Option<u8>
+}
+impl Cult{
+    pub fn on_phase_start(self, game: &mut Game, phase: PhaseType){
+        Cult::set_ordered_cultists(self.clone(), game);
+        
+        if phase == PhaseType::Night {
+            if self.can_convert_tonight(game){
+                game.add_message_to_chat_group(ChatGroup::Cult, ChatMessageVariant::ApostleCanConvertTonight);
+            }else{
+                game.add_message_to_chat_group(ChatGroup::Cult, ChatMessageVariant::ApostleCantConvertTonight);
+            }
+        }
+    }
+    pub fn on_game_start(self, game: &mut Game) {
+        Cult::set_ordered_cultists(self, game);
+    }
+    pub fn on_any_death(mut self, game: &mut Game){
+        self.sacrifices_needed = self.sacrifices_needed.map(|s| s.saturating_sub(1));
+        if let Some(s) = self.sacrifices_needed{
+            game.add_message_to_chat_group(ChatGroup::Cult, ChatMessageVariant::CultSacrificesRequired { required: s });
+        }
+        game.set_cult(self.clone());
+
+        Cult::set_ordered_cultists(self.clone(), game);
+    }
+    pub fn on_role_switch(self, game: &mut Game, actor: PlayerReference) {
+        if actor.role(game).faction() == Faction::Cult {
+            Cult::set_ordered_cultists(self.clone(), game);
+        }
+
+        for a in Cult::get_members(game) {
+            for b in Cult::get_members(game) {
+                a.insert_role_label(game, b, b.role(game));
+                b.insert_role_label(game, a, a.role(game));
+            }
+        }
+    }
+    pub fn get_members(game: &Game)->Vec<PlayerReference>{
+        PlayerReference::all_players(game).filter(
+            |p| p.role(game).faction() == Faction::Cult
+        ).collect()
+    }
+    pub fn get_living_members(&self, game: &Game)->Vec<PlayerReference>{
```

Comment:
Unused self parameter. Turn on your linter. Also, can't you just use ordered_cultists for this?

Reply:
removed it anyway